kong/init.lua

```diff
@@ -795,7 +812,7 @@ end
 
 
 function Kong.exit_worker()
-  if kong.configuration.role ~= "control_plane" and ngx.worker.id() == 0 then
+  if process.type() ~= "privileged agent" and kong.configuration.role ~= "control_plane" and worker_id() == 0 then
```

Comment:
Is there any reason we don't use privileged workers to manage plugin servers?

Reply:
@StarlightIbuki I wanted to skip it for now as I don't understand all the consequences of it. E.g. is it ok to have plugin servers managed with master process rights. I suggest we do this in steps. We haven't used privileged workers before, so making too many changes at once feels too risky. We can come back to that later.